### PR TITLE
#167733826 setup endpoints to assign roles and permissions

### PR DESCRIPTION
**Pull request by Ileriayo**
#### What does this PR do?
- It creates endpoints for a super administrator to:
i) assign permissions to roles and
ii) assign roles to users

#### Description of Task to be completed?
- Create endpoint to assign roles to users
- Create endpoint to assign permissions to users
- Create a middleware to check a user's role and grant permission to particular resource endpoints if authorized
- Create a model and migration file for the permissions table; permissions table will contain resource, role and permissions
- Create a model and migration file for the role table
- Create a seed file to seed roles  to the database
- Create a seed file to seed permissions to the database
- Create a seed file to seed a super admin to the database

#### How should this be manually tested?
- Send a `PATCH` request to the endpoint `/role` with a request body containing the user's email and the preferred role; e.g.
```
{
  "email": "user@barefootnomad.com",
  "role": "manager"
}
```
- Send a `PATCH` request to the endpoint `/role/permissions` with a request body containing the user's email and the preferred role; e.g.
```
{
  "role": "requester",
  "resource": "accomodation",
  "create": false,
  "read": true,
  "update": false,
  "delete": false
}
```

#### Any background context you want to provide?
- The super administrator will be required to assign the permissions a specific role should have against a particular resource endpoint such that a user with that role will be able to perform certain operations resource-wide and not app-wide. For instance, the super administrator should be able to assign permissions only a get (i.e., read = true) permission to 'requester' role for an `accomodation` endpoint (resource) but for a trip endpoint (resource), the super administrator should able to assign the following permissions - post (create = true), get ( read = true), patch/put ( update = true). The example shows permissions resource-wide against app-wide.

#### What are the relevant pivotal tracker stories?
- [#167733826](https://www.pivotaltracker.com/story/show/167733826)

#### Screenshots (if appropriate)
- N/A

### CODE REVIEW

File: src/database/migrations/20190820122711-create-user.js [hunk 1]
```diff
@@ -49,9 +49,24 @@ module.exports = {
         'travel_team_member',
         'manager',
         'requester',
+        'supplier',
         'user'
       ),
       defaultValue: 'user'
+      // validate: {
```
Comment (Oluwafayokemi) [inline src/database/migrations/20190820122711-create-user.js]: Remove commented code

File: src/docs/swagger.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ module.exports = {
     'http'
   ],
   // Uncomment for testing purposes
```
Comment (Oluwafayokemi) [inline src/docs/swagger.js]: Remove connected code

File: src/helpers/isPermitted.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+// import model from '../models';
```
Comment (Oluwafayokemi) [inline src/helpers/isPermitted.js]: Remove commented code

Comment (Ileriayo) [reply src/helpers/isPermitted.js]: Thanks for spotting this.
Done!

File: src/database/migrations/20190820122711-create-user.js (hunk 1 above)
Comment (Ileriayo) [reply src/database/migrations/20190820122711-create-user.js]: Thanks for spotting this.
Done!

File: src/docs/swagger.js (hunk 2 above)
Comment (Ileriayo) [reply src/docs/swagger.js]: Done!

**Review by EziamakaNV — APPROVED** [review]
Very well done!

**Review by Oluwafayokemi — APPROVED** [review]

**Review by Oluwafayokemi — APPROVED** [review]

**Review by Oluwafayokemi — APPROVED** [review]

**Pull request merged**